source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -64,13 +66,19 @@ struct SuccessResponse {
             Http::LowerCaseString{std::string(header.key().getStringView())},
             std::string(header.value().getStringView()));
       }
+      if (dynamic_metadata_matchers_->matches(header.key().getStringView())) {
+        std::string key{header.key().getStringView()};
+        std::string value{header.value().getStringView()};
+        (*response_->dynamic_metadata.mutable_fields())[key] = ValueUtil::stringValue(value);
+      }
       return Http::HeaderMap::Iterate::Continue;
     });
   }
 
   const Http::HeaderMap& headers_;
   const MatcherSharedPtr& matchers_;
   const MatcherSharedPtr& append_matchers_;
+  const MatcherSharedPtr& dynamic_metadata_matchers_;
```

Comment:
Sorry, can we rename this to `to_dynamic_metadata_matchers_` or similar, `headers_to_dynamic_metadata_matchers_`? A bit confused if we look at only this. Probably add a comment that all of these matchers (`matchers`, `append_matchers`, ...) work against `headers_`.

Reply:
good idea, this code is indeed getting hairy

Reply:
I went with `to_dynamic_metadata_matchers`